main.tf

```diff
@@ -173,7 +173,8 @@ module "ecs_alb_service_task" {
   subnet_ids                         = var.ecs_private_subnet_ids
   container_port                     = var.container_port
   nlb_container_port                 = var.nlb_container_port
-  docker_volumes                     = var.volumes
+  docker_volumes                     = [for volume in var.volumes : volume if length(volume.docker_volume_configuration) > 0]
+  efs_volumes                        = [for volume in var.volumes : volume if length(volume.efs_volume_configuration) > 0]
```

Comment:
@kevcube I suppose someone could do that, and it would not be a problem. `ecs_task_definition` does not seem to make these two mutually exclusive, check [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_task_definition#volume).

Reply:
Ok if ECS allows it, but every volume would be configured as both a docker volume and efs volume in that case. I'm unsure if this will cause any issue.

Reply:
Hi @kevcube .. Let's take the docker case as an example. This `volume if length(volume.docker_volume_configuration) > 0`: This is a condition that filters the list. The volume will only be included in the final list if the condition length(volume.docker_volume_configuration) > 0 is true. 

The end result of this loop is a list that contains only those volumes from var.volumes where the `docker_volume_configuration` attribute is not empty. And the same happens for EFS volumes. So we're basically splitting the input array, the docker volumes become their own list, and the EFS ones become their own list. So I don't think there is any trouble with this.

Reply:
Ah I see, I misread the `volume.` as `var.` and didn't realize this was in a loop. Ok then I think this is good to merge!